modules/caddyhttp/metrics.go

```diff
@@ -0,0 +1,215 @@
+	"github.com/prometheus/client_golang/prometheus/promauto"
+)
+
+var httpMetrics = struct {
+	init                   sync.Once
+	requestInFlight        *prometheus.GaugeVec
+	requestCount           *prometheus.CounterVec
+	requestErrors          *prometheus.CounterVec
+	requestDuration        *prometheus.HistogramVec
+	requestDurationSummary *prometheus.SummaryVec
+	requestSize            *prometheus.HistogramVec
+	responseSize           *prometheus.HistogramVec
+	responseLatency        *prometheus.HistogramVec
+	responseLatencySummary *prometheus.SummaryVec
+}{
+	init: sync.Once{},
+}
+
+func initHTTPMetrics() {
+	ns := "caddy"
+	sub := "http"
+
+	basicLabels := []string{"server", "handler"}
+	httpMetrics.requestInFlight = promauto.NewGaugeVec(prometheus.GaugeOpts{
+		Namespace: ns,
+		Subsystem: sub,
+		Name:      "requests_in_flight",
+		Help:      "Number of requests currently handled by this server.",
+	}, basicLabels)
+	httpMetrics.requestErrors = promauto.NewCounterVec(prometheus.CounterOpts{
+		Namespace: ns,
+		Subsystem: sub,
+		Name:      "request_errors_total",
+		Help:      "Number of requests resulting in middleware errors.",
+	}, basicLabels)
+	httpMetrics.requestCount = promauto.NewCounterVec(prometheus.CounterOpts{
+		Namespace: ns,
+		Subsystem: sub,
+		Name:      "requests_total",
+		Help:      "Counter of HTTP(S) requests made.",
+	}, basicLabels)
+
+	// TODO: allow these to be customized in the config
+	latencyBuckets := []float64{.01, .05, .1, .2, .4, 1, 3, 8, 20, 60, 120}
+	sizeBuckets := prometheus.ExponentialBuckets(256, 4, 8)
+	quantileBuckets := map[float64]float64{0.1: 0.01, 0.5: 0.01, 0.95: 0.01, 0.99: 0.001, 0.999: 0.0001}
+
+	httpLabels := []string{"server", "handler", "code", "method"}
+	httpMetrics.requestDuration = promauto.NewHistogramVec(prometheus.HistogramOpts{
+		Namespace: ns,
+		Subsystem: sub,
+		Name:      "request_duration_seconds",
+		Help:      "Histogram of round-trip request durations.",
+		Buckets:   latencyBuckets,
+	}, httpLabels)
+	httpMetrics.requestDurationSummary = promauto.NewSummaryVec(prometheus.SummaryOpts{
+		Namespace:  ns,
+		Subsystem:  sub,
+		Name:       "request_duration_quantile_seconds",
+		Help:       "Summary of round-trip request durations.",
+		Objectives: quantileBuckets,
+	}, httpLabels)
+	httpMetrics.requestSize = promauto.NewHistogramVec(prometheus.HistogramOpts{
+		Namespace: ns,
+		Subsystem: sub,
+		Name:      "request_size_bytes",
+		Help:      "Total size of the request. Includes body",
+		Buckets:   sizeBuckets,
+	}, httpLabels)
+	httpMetrics.responseSize = promauto.NewHistogramVec(prometheus.HistogramOpts{
+		Namespace: ns,
+		Subsystem: sub,
+		Name:      "response_size_bytes",
+		Help:      "Size of the returned response.",
+		Buckets:   sizeBuckets,
+	}, httpLabels)
+	httpMetrics.responseLatency = promauto.NewHistogramVec(prometheus.HistogramOpts{
+		Namespace: ns,
+		Subsystem: sub,
+		Name:      "response_latency_seconds",
```

Comment:
This should be `response_duration_seconds`.

Reply:
Ah - this was a holdover from [the v1 module](https://github.com/miekg/caddy-prometheus/blob/master/metrics.go#L63) - I can change this.

```suggestion
		Name:      "response_duration_seconds",
```